x/evm/keeper/params.go

```diff
@@ -23,6 +24,10 @@ func (k Keeper) GetParams(ctx sdk.Context) (params types.Params) {
 
 // SetParams sets the EVM params each in their individual key for better get performance
 func (k Keeper) SetParams(ctx sdk.Context, params types.Params) error {
+	// NOTE: We need to sort the precompiles in order to enable searching with binary search
+	// in params.IsActivePrecompile.
+	slices.Sort(params.ActivePrecompiles)
+
```

Comment:
Since the `IsActivePrecompile` method of the Params uses binary search, I've added this instruction to sort the slice of active precompiles when setting the params. Also added corresponding tests.